src/unleash_client_manager.ts

```diff
@@ -9,13 +9,58 @@ import { registerStrategies } from "./strategy/appStudioStrategies";
 // map value = Unleash client
 const unleashClientMap = new Map<string, Unleash>();
 
+export interface InitializeAttempt {
+  numAttempt: number;
+  timeAttempt: number;
+  isBlocked: boolean;
+}
+
+// map key = extensionName
+// map value = number of attempts to establish unleash client
+const initializeAttemptMap = new Map<string, InitializeAttempt>();
+const limitNumber = 2;
+
+function handleUnauthorisedError(extensionName: string): void {
+  let num = initializeAttemptMap.get(extensionName)?.numAttempt;
+  num = (num || 0) + 1;
+  if (num < limitNumber) {
+    initializeAttemptMap.set(extensionName, { numAttempt: num, timeAttempt: 0, isBlocked: false });
+  }
+  if (num == limitNumber) {
+    initializeAttemptMap.set(extensionName, { numAttempt: num, timeAttempt: Date.now(), isBlocked: true });
+  }
```

Comment:
I would try to change it to:
```js 
let isInitBlocked = num == limitNumber ? true : false;
initializeAttemptMap.set(extensionName, { numAttempt: num, timeAttempt: Date.now(), isBlocked: isInitBlocked });
```

Reply:
Theoretically, I didnt want to modify this map  after limit is reached, but on the other side this code should not be reachable after limit is reached. Did this change. :)